### DISCUSSION TITLE
The R utility to read files from the underlying ESP32 host disk is not working under CDOS 258 (or Cromix)

### DISCUSSION THREAD

**Discussion by riwide** [thehighnibble/thehighnibble.github.io#61] [Cromemco Z-1 replica]
I am trying to use R to pull files from the underlying ESP32 SD fat disk, but CDOS 258 is only creating 0 length files on the target drive. Tried with both floppy and hd as the target disk.

I assume W may have similar problems.

The R.COM program is on the CDOS258 image so I assumed it should be supported on the Z1 replica.

I wish to resolve file transfer into the system with a convenient method.  On my other systems I am happy with kermit or xmodem, so  I thought I'd use R to pull in a version of kermit, then use that to transfer files over a serial port.

**Comment by pfcushing** [comment]
Some time ago, I asked Udo Munk about these two utilities and he replied that they both work correctly on the Z80PACK running under Windows/Cygwin because he included the BDOS extension code in the z80pack, which is necessary to allow access by these utilities.  Since the Imsaisim and Cromemcosim  are backports of z80pack, rewritten for the ESP32, one would need to know whether Dave included this extension in his backport.

Paul Cushing

**Comment by riwide** [comment]
Ok, thanks. Helpful comment. I've emailed David McNaughton to find out more.

Richard

**Comment by udo-munk** [comment]
The BDOS emulation required to access the host filesystem with the R & W commands is not implemented in the replicas, can't be used there, only works with z80pack running on some PC.

**Comment by riwide** [comment]
Ok, thanks.
